synchronizer/synchronizer.go

```diff
@@ -236,6 +232,58 @@ func (s *ClientSynchronizer) syncBlocks(lastEthBlockSynced *state.Block) (*state
 	return lastEthBlockSynced, nil
 }
 
+// syncTrustedState synchronizes information from the trusted sequencer
+// related to the trusted state when the noda has all the information from
+// l1 synchronized
+func (s *ClientSynchronizer) syncTrustedState(latestSyncedBatch uint64) error {
+	if s.broadcastClient == nil {
+		return nil
+	}
+
+	log.Info("Getting trusted state info")
+	lastTrustedStateBatch, err := s.broadcastClient.GetLastBatch(s.ctx, &emptypb.Empty{})
+	if err != nil {
+		log.Warn("error syncing trusted state. Error: ", err)
+		return err
+	}
+
+	log.Debug("lastTrustedStateBatch.BatchNumber ", lastTrustedStateBatch.BatchNumber)
+	log.Debug("latestSyncedBatch ", latestSyncedBatch)
+	if lastTrustedStateBatch.BatchNumber < latestSyncedBatch {
+		return nil
+	}
+
+	batchNumberToSync := latestSyncedBatch
+	for batchNumberToSync <= lastTrustedStateBatch.BatchNumber {
+		batchToSync, err := s.broadcastClient.GetBatch(s.ctx, &pb.GetBatchRequest{BatchNumber: batchNumberToSync})
+		if err != nil {
+			log.Warnf("failed to get batch %v from trusted state via broadcast. Error: %v", batchNumberToSync, err)
+			return err
+		}
+
+		dbTx, err := s.state.BeginStateTransaction(s.ctx)
+		if err != nil {
+			log.Fatalf("error creating db transaction to sync trusted batch %v: %v", batchNumberToSync, err)
+		}
+
+		if err := s.processTrustedBatch(batchToSync, dbTx); err != nil {
+			log.Fatalf("error processing trusted batch %v: %v", batchNumberToSync, err)
+			err := dbTx.Rollback(s.ctx)
```

Comment:
This line is unreachable, as the previous line will throw a panic

Reply:
replaced this `Fatal` by `Error`